### PR TITLE
Run3-hcx231 Preparing to remove accessing DDDCompactView from SimG4CMS/Calo code (Step 1)

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Access spectra parameters to be used for simulation in the Geometry step and make it available for SIM so that SIM will not deal with DDDCompactView.

#### PR validation:

Use standard runTheMatrix.py

#### if this PR is a backport please specify the original PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: CondFormats/GeometryObjects/src/classes_def.xml [hunk 1]
```diff
@@ -167,14 +167,30 @@
    <field name="copiesInLayers_"  mapping="blob"/>
  </class>
  <class name="HcalSimulationParameters" class_version="0">
```
Comment (ianna) [inline CondFormats/GeometryObjects/src/classes_def.xml]: shouldn't it be a new version?

File: Geometry/HcalCommonData/interface/CaloDDDSimulationConstants.h [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ */
+
+#include "CondFormats/GeometryObjects/interface/CaloSimulationParameters.h"
+
+class CaloDDDSimulationConstants {
+public:
+  CaloDDDSimulationConstants(const CaloSimulationParameters* cps);
+  ~CaloDDDSimulationConstants();
+
+  const CaloSimulationParameters* claosimpar() const { return calospar_; }
```
Comment (cvuosalo) [inline Geometry/HcalCommonData/interface/CaloDDDSimulationConstants.h]:
```suggestion
  const CaloSimulationParameters* caloSimPar() const { return calospar_; }
```

**Comment by cvuosalo** [comment]
The title of the PR would be clearer if it were "Preparing to remove..." rather than "Trying to remove...". None of the files in this PR are from SimG4CMS.

File: CondFormats/GeometryObjects/src/classes_def.xml (hunk 1 above)
Comment (bsunanda) [reply CondFormats/GeometryObjects/src/classes_def.xml]: Changed

File: Geometry/HcalCommonData/interface/CaloDDDSimulationConstants.h (hunk 2 above)
Comment (bsunanda) [reply Geometry/HcalCommonData/interface/CaloDDDSimulationConstants.h]: Done

File: CondFormats/GeometryObjects/src/classes_def.xml (hunk 1 above)
Comment (Dr15Jones) [reply CondFormats/GeometryObjects/src/classes_def.xml]: FYI, a class_version == 0 tells ROOT not to store that object.

Comment (bsunanda) [reply CondFormats/GeometryObjects/src/classes_def.xml]: Should I change it back to 0 then?

**Comment by cvuosalo** [comment]
+1

**Comment by ggovi** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**